compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4175,19 +4175,23 @@ private BLangMatchPattern transformMatchPattern(Node matchPattern) {
                     (BLangListMatchPattern) TreeBuilder.createListMatchPattern();
             bLangListMatchPattern.pos = matchPatternPos;
 
-            for (Node memberMatchPattern : listMatchPatternNode.matchPatterns()) {
-                BLangMatchPattern bLangMemberMatchPattern = transformMatchPattern(memberMatchPattern);
+            SeparatedNodeList<Node> matchPatterns = listMatchPatternNode.matchPatterns();
+            int matchPatternListSize = matchPatterns.size();
+            for (int i = 0; i < matchPatternListSize - 1; i++) {
```

Comment:
Is it intended to iterate only up to `matchPatternListSize - 1`, eliminating the last one? 
How about having a `foreach` instead of  `for (int i = 0; i < matchPatternListSize - 1; i++)` ?